parsl/executors/high_throughput/interchange.py

```diff
@@ -282,10 +285,27 @@ def migrate_tasks_to_internal(self, kill_event):
                 task_counter += 1
                 logger.debug("[TASK_PULL_THREAD] Fetched task:{}".format(task_counter))
 
+    def _send_monitoring_info(self, manager):
+        if self.monitoring_enabled:
+            logger.info("Sending message {} to hub".format(self._ready_manager_queue[manager]))
+            self.hub_channel.send_pyobj((MessageType.NODE_INFO,
+                                         datetime.datetime.now(),
+                                         self._ready_manager_queue[manager]))
+
     def _command_server(self, kill_event):
         """ Command server to run async command to the interchange
         """
         logger.debug("[COMMAND] Command Server Starting")
+
+        # Need to create a new ZMQ socket for command server thread
+        monitoring_enabled = False
```

Comment:
This block means that each manager will have a two monitoring TCP connections to the monitoring hub; and the second connection will only be used in the case of HOLD_WORKER commands.

I feel uncomfortable with that extra load on the hub side of things: this would be thousands of extra connections in the case of LSST DRP, for example.

Maybe there's a way to send that shutdown message from the main thread, or a similar message - for example, combined with the code that reports `bad_managers` to monitoring?

Reply:
After a meeting with @ZhuozhaoLi, I realise this is only one connection per interchange, not per note, so I withdraw this object.